src/honey_badger/builder.rs

```diff
@@ -35,6 +38,7 @@ where
     pub fn new(netinfo: Arc<NetworkInfo<N>>) -> Self {
         HoneyBadgerBuilder {
             netinfo,
+            session_id: 0,
```

Comment:
Is it ever a good idea to have a slightly insecure default here? We could make it a mandatory argument and force people to pass in `0` explicitly. What's the most common use case?

Reply:
I'm not sure: As soon as you generate new key shares every time you start Honey Badger, you're fine with session ID 0. If you restart a network, but in a new epoch, you're also fine.

And even if you restart the exact same network at epoch 0, all the attacker could do would be delay consensus slightly more effectively than in the first run: If e.g. in epoch 0, one of the coins was flipped twice, they'll now know two coin values in advance, but from the third one, they won't, so they won't be able to stall consensus indefinitely anyway.

I'm fine with either: making it mandatory or not.

Reply:
I'm not 100% sure about this either but I think it's probably better left with a 0 default.